File: leetcode/go/120.triangle.go
func minimumTotal(triangle [][]int) int {
    for row := 1; row < len(triangle); row++ {
        triangle[row][0] += triangle[row-1][0]
        for col := 1; col < row ; col++ {
            triangle[row][col] += Min(triangle[row-1][col-1], triangle[row-1][col])
        }
        triangle[row][row] += triangle[row-1][row-1]
    }
    min_path := 1000000
    lastRow := triangle[len(triangle)-1]
    for _, path := range lastRow {
        min_path = Min(min_path, path)
    }
    return min_path
}

func Min(a int, b int) int {
    if a < b {return a}
    return b
}

File: leetcode/rust-leetcode/src/120.triangle.rs
use std::cmp::min;
impl Solution {
    pub fn minimum_total(mut triangle: Vec<Vec<i32>>) -> i32 {
        for row in 1..triangle.len() {
            for col in 0..=row {
                triangle[row][col] += min(triangle[row-1][col.saturating_sub(1)], triangle[row-1][col.min(row-1)]);
            }
        }
        *triangle.last().unwrap().iter().min().unwrap()
    }
}

